### DISCUSSION TITLE
Teensy design start

### DISCUSSION THREAD

**Discussion by PumpkinClouds** [Comet-Robotics/URC#21] [PCB/Power]
I found the Teensy 4.1 board in this github discussion: https://github.com/jenschr/Teensy-4.1-example
it might be helpful to look up how to import each of the files but it has examples and importing the schematic is easy enough just download the Mim .kicad_sym thing and go to preferences and symbol library -> folder icon and select it. do more research on how to get the rest

**Comment by pablo-collantes** [comment]
Alright so I ended up downloading two github repo's:

## First

Download the [Teensy Library repo](https://github.com/XenGi/teensy_library) here by just downloading it all into a .zip and extracting it anywhere doesnt matter. This is going to be our schematic and overall symbol library.

Download the [teensy.pretty repo](https://github.com/XenGi/teensy.pretty?tab=readme-ov-file) here by also downloading into .zip and extracting anywhere. This is our footprint and 3d model library for when you enter the PCB editor mode.

## Second 

Now you have to add these libraries to your kicad
you can add the teensy library file first by just being in the project viewer like this:
<img src="https://github.com/user-attachments/assets/47fd8fc8-f185-4a95-b014-4098c0684c64" width= 500>

Then click:
**Preferences** -> **Manage Symbol Libraries** 
Then click this little folder icon in the bottom left:
<img src="https://github.com/user-attachments/assets/6194d21b-39da-4a86-b266-e0f64e874c71" width= 400>
Navigate to your **teensy_library-master** folder and select the **teensy.lib** file
You can now select the new row that it added nicknamed teensy and click the Migrate Libraries button so that you can convert the file from a legacy format to a kicad_sym format.

## Third

Basically you're going to do the exact same thing but now with the footprint
So once again click:
**Preferences** -> **Manage Footprint Libraries**
Click the little folder icon and navigate through the **teensy.pretty-master** folder until theres nothing left because for footprints it wants the entire folder not just a file.

And you should be done, if you created a teensy project file already you can go into PCB Editor mode and click 
**Place** -> **Add Footprint**
And you just have to filter for teensy and select the teensy41 and it should look like this:

In PCB Editor Mode:
<img src="https://github.com/user-attachments/assets/9a47f963-935d-4223-ad23-df8b709fa103" width=500>

For schematic mode i imported the symbol by clicking:
**View** -> **Symbol Library Browser** and just searched for it in there and placed it
In Symbol/Schematic Editor Mode:
<img src="https://github.com/user-attachments/assets/5ac15e31-8230-4e6c-99b8-70194ad3678f" width=500>
